integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -435,14 +436,16 @@ class CarbonScanRDD[T: ClassTag](
               "true")
             if (carbonRecordReader == null) {
               new CarbonRecordReader(model,
-                format.getReadSupportClass(attemptContext.getConfiguration), inputMetricsStats)
+                format.getReadSupportClass(attemptContext.getConfiguration),
+                inputMetricsStats,
+                getConf)
             } else {
               carbonRecordReader
             }
           } else {
             new CarbonRecordReader(model,
               format.getReadSupportClass(attemptContext.getConfiguration),
-              inputMetricsStats)
+              inputMetricsStats, getConf)
```

Comment:
In the executor side avoid calling getConf  multiple times. Instead we can take from FileFactory.